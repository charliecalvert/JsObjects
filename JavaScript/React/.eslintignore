**/registerServiceWorker.js
**/serviceWorker.js
**/bundle.js
.git
**/node_modules/**
**/bower_components/**
**/coverage/**
**/build/**
**/static/**
**/elf-gatsby-site/**
**/manifest.json
**/precache-manifest.*.js
**/service-worker.js
ReactEs6Browserify/public/javascripts/basics.js
MaterialBrowserify/public/javascripts/basics.js
