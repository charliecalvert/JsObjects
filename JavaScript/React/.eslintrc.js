module.exports = {
    env: {
        browser: true,
        es2021: true,
        'jest/globals': true,
    },
    extends: [
        'plugin:react/recommended',
        'airbnb',
    ],
    parserOptions: {
        ecmaFeatures: {
            jsx: true,
        },
        ecmaVersion: 'latest',
        sourceType: 'module',
    },
    plugins: [
        'react',
        'jest',
    ],
    rules: {
        indent: ['error', 4],
        'react/jsx-indent': ['error', 4],
        'react/jsx-indent-props': ['error', 4],
        'no-console': 'off',
    },
};
