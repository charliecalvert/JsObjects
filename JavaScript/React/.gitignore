foo
basic
food-lookup-demo
!components/
bundle.js
