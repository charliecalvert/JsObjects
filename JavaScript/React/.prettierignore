**/build/**
**/serviceWorker.js
**/bower_components/**
**/materialize.min.js
**/gatsby-site/**
**/coverage/**
ReactEs6Browserify/public/javascripts/basics.js
MaterialBrowserify/public/javascripts/basics.js
