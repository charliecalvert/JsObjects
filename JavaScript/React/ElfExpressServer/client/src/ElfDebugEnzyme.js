/**
 * Created by charlie on 5/7/17.
 *
 * Initialize it like this:
 *
 *   import ElfDebugEnzyme from '../ElfDebugEnzyme';
 *   const elfDebugEnzyme = new ElfDebugEnzyme(true, 'Foo.test.js');
 *
 * And then use it:
 *
 *   elfDebugEnzyme.getIndex(wrapper, 'div#addressShow', 3, false);
 */

export default class ElfDebugEnzyme {
    /**
     * @param showData: Whether or not to display output
     *
     * @param callerName: Pass in the name of the class that is
     * using this debug tool. That way, each message written to
     * the console includes the name of the class that is
     * requesting to see the output.
     */
    constructor(showData = false, callerName = '') {
        this.showData = showData;
        this.callerName = callerName + ':\n';
    }

    display(value) {
        console.log(this.callerName + value);
    }

    getAll(wrapper, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper.debug();
            this.display(paragraphData);
        }
    }

    getElement(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper.find(element).debug();
            this.display(paragraphData);
        }
    }

    getFirst(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper
                .find(element)
                .first()
                .debug();
            this.display(paragraphData);
        }
    }

    getIndex(wrapper, element, index, showMe) {
        if (this.showData || showMe) {
            var paragraphData = wrapper
                .find(element)
                .childAt(index)
                .debug();
            this.display(paragraphData);
        }
    }

    getLast(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper
                .find(element)
                .last()
                .debug();
            this.display(this.callerName + paragraphData);
        }
    }
}
