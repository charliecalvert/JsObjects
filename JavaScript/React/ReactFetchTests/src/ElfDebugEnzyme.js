/**
 * Created by charlie on 5/7/17.
 *
 * Initialize it like this:
 *
 *   import ElfDebugEnzyme from '../ElfDebugEnzyme';
 *   const elfDebugEnzyme = new ElfDebugEnzyme(true, 'Foo.test.js');
 *
 * And then use it in a test:
 *
 *   elfDebugEnzyme.getIndex(wrapper, 'div#addressShow', 3, false);
 *
 * If you pass in true to the constructor then you always see the output from this code.
 * If you pass false to the constructor, but true to one of the individual methods, then
 * you see the output from only that method. For instance, the true in the call to getLast
 * overrides the false in the constructor:
 *
 *   const elfDebugEnzyme = new ElfDebugEnzyme(false, 'App.test.js');
 *
 *   it('renders and reads H1 text', () => {
 *      const wrapper = shallow(<App />);
 *      const welcome = <h1>Waiting</h1>;
 *      elfDebugEnzyme.getLast(wrapper, 'h1', true);
 *      expect(wrapper.contains(welcome)).toEqual(true);
 *   });
 *
 */

export default class ElfDebugEnzyme {
    /**
     * @param showData: Whether or not to display output
     *
     * @param callerName: Pass in the name of the class that is
     * using this debug tool. That way, each message written to
     * the console includes the name of the class that is
     * requesting to see the output.
     */
    constructor(showData = false, callerName = '') {
        this.showData = showData;
        this.callerName = callerName + ':\n';
    }

    display(value) {
        console.log(this.callerName + value);
    }

    getAll(wrapper, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper.debug();
            this.display(paragraphData);
        }
    }

    getAllDive(wrapper, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper.dive().debug();
            this.display(paragraphData);
        }
    }

    getElement(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper.find(element).debug();
            this.display(paragraphData);
        }
    }

    getFirst(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper
                .find(element)
                .first()
                .debug();
            this.display(paragraphData);
        }
    }

    getIndex(wrapper, element, index, showMe) {
        if (this.showData || showMe) {
            var paragraphData = wrapper
                .find(element)
                .childAt(index)
                .debug();
            this.display(paragraphData);
        }
    }

    getLast(wrapper, element, showMe) {
        if (this.showData || showMe) {
            const paragraphData = wrapper
                .find(element)
                .last()
                .debug();
            this.display(this.callerName + paragraphData);
        }
    }
}
