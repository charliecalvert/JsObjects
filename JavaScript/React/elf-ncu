#! /usr/bin/env bash

RED='\033[0;31m'
LIGHT_RED='\033[1;31m'
LIGHT_GREEN='\033[1;32m'
YELLOW='\033[1;33m'
BLUE='\033[1;36m'
NC='\033[0m' # No Color

START_DIR=$HOME'/Git/JsObjects/JavaScript/React/'

function check-return-value {
    retVal=$?
    if [ $retVal -ne 0 ]; then
        echo -e ${LIGHT_RED}"Error"${NC}
        ncu -u
        rm -rf node_modules/ && rm package-lock.json && npm i
    else
        echo -e ${LIGHT_GREEN}"Ok"${NC}
    fi
}

function spacer {
    echo -e ${YELLOW}======================${NC}
    echo -e ${YELLOW} $1 ${NC}
    echo -e ${YELLOW}======================${NC}
}

function check-dir {
    spacer $1
    cd $START_DIR
    cd $1
    pwd
    ncu -e2    
    check-return-value
}

function get-directories-containing-package-json {
    FindNp package.json | xargs -I{} dirname {} | xargs printf "check-dir %s\n" | sort
}

check-dir ./drawer-menu
check-dir ./ElfExpressServer
check-dir ./ElfExpressServer/client
check-dir ./ElfExpressServer/server
check-dir ./higher-order-component
check-dir ./MaterialBrowserify
check-dir ./ReactEs6/ReactBasics
check-dir ./ReactEs6/ReactEs6Bower
check-dir ./ReactEs6/ReactEs6Browserify
check-dir ./ReactFetchTests
#check-dir ./react-native-test
#check-dir ./ReactNativeTesting/EnzymeBasics
#check-dir ./ReactNativeTesting/SnapShotBasics
check-dir ./react-simple
check-dir ./react-webpack-simple
check-dir ./redux-simple
check-dir ./RestBoiler
check-dir ./RestExpress
check-dir ./material-button-components
check-dir ./simple-radio-buttons
check-dir ./simple-state
check-dir ./style-guide-examples
check-dir ./todo-redux
#check-dir ./gatsby-site
