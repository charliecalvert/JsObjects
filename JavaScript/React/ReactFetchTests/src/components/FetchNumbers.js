import React, { Component } from 'react';
import '../css/App.css';

class FetchServerInfo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            numbers: ['a', 'b'],
            error: 'none'
        };

        this.fetchNumbers = this.fetchNumbers.bind(this);
    }

    fetchNumbers() {
        const that = this;
        return fetch('/fetch-tests/numbers')
            .then(response => response.json())
            .then(function(result) {
                console.log(result);
                that.setState({
                    numbers: result.numbers,
                    error: 'none'
                });
            })
            .catch(function(ex) {
                console.log('ERROR', ex.message);
                console.log('ERROR', ex.stack);
                console.log('ERROR', JSON.stringify(ex, null, 4));
                that.setState({ error: ex.message });
            });
    }

    render() {
        console.log('RENDER FETCH NUMBERS', this.state);
        return (
            <div className="App">
                {this.state.numbers.map(value => {
                    return <p key={value}>{value}</p>;
                })}
                <p className="App-intro">Error: {this.state.error}</p>
                <button id="fetchNumbers" onClick={this.fetchNumbers}>
                    Fetch Numbers
                </button>

                <hr />
            </div>
        );
    }
}

export default FetchServerInfo;
