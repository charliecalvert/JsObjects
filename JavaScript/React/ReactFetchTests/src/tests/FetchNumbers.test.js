import React from 'react';
import ReactDOM from 'react-dom';
import FetchNumbers from '../components/FetchNumbers';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ElfDebugEnzyme from '../ElfDebugEnzyme';
import elfLogger from '../elf-logger';

const elfDebugEnzyme = new ElfDebugEnzyme(false, 'App.test.js');
configure({ adapter: new Adapter() });

describe('FetchNumbers Tests', function() {
    elfLogger.off();

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FetchNumbers />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('handles click on foobar with enzyme', () => {
        const wrapper = shallow(<FetchNumbers />);
        console.log(wrapper.debug());
        wrapper.find('#fetchNumbers').simulate('click');
        expect(wrapper).toMatchSnapshot();
    });

    it('renders state.result before button click', () => {
        const wrapper = shallow(<FetchNumbers />);
        const nineSign = <p>a</p>;
        elfDebugEnzyme.getFirst(wrapper, 'p');
        expect(wrapper.contains(nineSign)).toBe(true);
    });

    it('renders state by calling fetchNumbers directly', () => {
        const wrapper = shallow(<FetchNumbers />);
        const statusParagraph = <p>One</p>;
        wrapper
            .instance()
            .fetchNumbers()
            .then(() => {
                wrapper.update();
                console.log(wrapper.debug());
                expect(wrapper.contains(statusParagraph)).toBe(true);
            });
    });

    it('calls fetchNumbers button click', () => {
        const spy = jest.spyOn(FetchNumbers.prototype, 'fetchNumbers');
        const wrapper = shallow(<FetchNumbers />);
        wrapper.find('#fetchNumbers').simulate('click');
        console.log(wrapper.debug());
        expect(spy).toHaveBeenCalled();
    });

    it('renders state.result after button click with setImmediate', () => {
        const wrapper = shallow(<FetchNumbers />);
        const statusParagraph = <p>One</p>;
        wrapper.find('#fetchNumbers').simulate('click');
        setTimeout(() => {
            wrapper.update();
            elfDebugEnzyme.getFirst(wrapper, 'p');
            try {
                expect(wrapper.contains(statusParagraph)).toBe(true);
            } catch (e) {
                console.error(e);
            }
        }, 0);
    });
});
