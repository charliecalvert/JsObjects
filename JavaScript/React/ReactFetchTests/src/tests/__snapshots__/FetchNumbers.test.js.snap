// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchNumbers Tests handles click on foobar with enzyme 1`] = `ShallowWrapper {}`;
